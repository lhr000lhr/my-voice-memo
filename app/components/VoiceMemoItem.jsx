import React from 'react'
import { connect } from 'react-redux'
import { Text } from 'react-native'
import { ListItem } from 'react-native-elements'

function VoiceMemoItem({ recording, onPress }) {
  return (
    <ListItem onPress={onPress} bottomDivider>
      <ListItem.Content>
        <Text>{recording?.uri}</Text>
      </ListItem.Content>
    </ListItem>
  )
}

export default connect()(VoiceMemoItem)
