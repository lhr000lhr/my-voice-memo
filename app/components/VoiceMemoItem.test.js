import React from 'react'
import renderer from 'react-test-renderer'
import VoiceMemoItem from './VoiceMemoItem'

jest.mock('moment', () => () => jest.requireActual('moment')('2020-01-01T00:00:00.000Z'))

describe('VoiceMemoItem', () => {
  it('renders correctly with VoiceMemoItem', () => {
    expect(renderer.create(<VoiceMemoItem />)).toMatchSnapshot()
  })
})
