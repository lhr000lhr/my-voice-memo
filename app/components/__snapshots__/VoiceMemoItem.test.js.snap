// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoiceMemoItem renders correctly with VoiceMemoItem 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": 0.5,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="padView"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      theme={
        Object {
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "searchBg": "#dcdce1",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "default": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
              "web": Object {
                "error": "#ff190c",
                "grey": "#393e42",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
      >
        
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#242424",
            "fontSize": 15,
          }
        }
        testID="listItemTitle"
      >
        01/01/2020
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            convert to text
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingLeft": 16,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
